Change wxBitmap to wxBitmapBundle for few controls

kosh543 opened the pull request.

kosh543 committed:
Change wxBitmap to wxBitmapBundle in wxBitmapToggleButton

kosh543 committed:
Change wxBitmap to wxBitmapBundle in wxBannerWindow

kosh543 committed:
Change inactive bitmap to in wxBitmapBundle wxAnimationCtrl

vadz commented on include/wx/animate.h:
```diff
@@ -19,6 +19,7 @@
 #include "wx/control.h"
 #include "wx/timer.h"
 #include "wx/bitmap.h"
```
```suggestion
```
(for consistency, we should remove existing inclusions of `wx/bitmap.h` when adding includes of `wx/bmpbndl.h` everywhere, not just in some places).

vadz commented on include/wx/bannerwindow.h:
```diff
@@ -14,11 +14,11 @@
 #include "wx/window.h"
 
-class WXDLLIMPEXP_FWD_CORE wxBitmap;
+class WXDLLIMPEXP_FWD_CORE wxBitmapBundle;
```
```suggestion
```
This is unnecessary (just as the old forward declaration was, but as we're changing this anyhow, it's better to remove it instead of preserving).

vadz commented on include/wx/generic/animate.h:
```diff
@@ -110,6 +110,15 @@ class WXDLLIMPEXP_ADV wxGenericAnimationCtrl: public wxAnimationCtrlBase
     virtual void DisplayStaticImage() wxOVERRIDE;
     virtual wxSize DoGetBestSize() const wxOVERRIDE;
 
+    // This function can be used as event handle for wxEVT_DPI_CHANGED event
```
```suggestion
    // This function can be used as event handler for wxEVT_DPI_CHANGED event
```

vadz reviewed (commented):
Thanks, no real comments here and we can already merge this (and I'll do it soon unless there are any objections), but we need to also update the various `Get{Width,Height}()` calls to use `GetScaledXXX()` instead.

vadz commented on src/generic/bannerwindow.cpp:
```diff
@@ -109,9 +109,9 @@ wxFont wxBannerWindow::GetTitleFont() const
+    if ( m_bitmapBundle.IsOk() )
     {
-        return m_bitmap.GetSize();
+        return m_bitmapBundle.GetPreferredSizeFor(this);
```
```suggestion
        return FromPhys(m_bitmapBundle.GetPreferredSizeFor(this));
```

vadz reviewed (commented):
Looks good except for a single missing conversion to physical, thanks!

kosh543 committed:
Rename wxBitmapBundle::GetPreferredSizeFor()

Rename wxBitmapBundle::GetPreferredSizeFor() to GetPreferredBitmapSizeFor to be more explicit
Add wxBitmapBundle::GetPreferredLogicalSizeFor() which return GetPreferredBitmapSizeFor() in logical size

kosh543 commented:
Uses #22056

vadz commented:
I've cherry-picked this as 3 separate commits to avoid merge conflicts after merging the other PR:
85becc93620dc7122c11cda79b875b6c2d220722, 2b94729a33db549569cd7802f0eef6b8afdd400d and 3f3561e2c697c06086de42a9a8e37552c7274645.
